### ISSUE TITLE
Separating haplotypes that occur at lower coverage

### ISSUE THREAD

**Issue by tcb72** [chhylp123/hifiasm#92]
Hello again,

Same genome as last time (diploid algae, 2% heterozygosity.) I've been playing around with both purging and not purging. Given the haplotypes are highly divergent, we get clear separation of haplotypes when Hi-C scaffolding the non-purged assembly (via Juicer/3D-DNA., see the following:)

![image](https://user-images.githubusercontent.com/29762255/113771543-ba74f580-96e0-11eb-9216-bb9dc7e19b8a.png)

In some areas of the genome, one of the haplotypes doesn't have as much coverage as the other. For example, if the coverage is 40x, there might be 4 reads which show a clear variant signal. This causes one of the chromosomes within a pair to be collapsed, as shown here:

![image](https://user-images.githubusercontent.com/29762255/113771927-3d964b80-96e1-11eb-92c4-e0eb12469b7b.png)

And here's that same area with variants highlighted in Tablet (the HiFi reads aligned with Minimap, -ax asm20 no-secondary):

![image](https://user-images.githubusercontent.com/29762255/113772406-e775d800-96e1-11eb-8ec2-71669da2d940.png)

Even more strange is that short-read alignment in that area shows about 50/50 coverage between the haplotypes, which is markedly different from the 90/10 coverage between haplotypes w/ the HiFi reads:

![image](https://user-images.githubusercontent.com/29762255/113772875-81d61b80-96e2-11eb-9e43-552cdf4d8b3a.png)

I'm not sure why the HiFi reads aren't capturing the variants to the degree the short-reads are, but there's clearly some signal there in the HiFi alignment.

With respect to hifiasm, are there any parameters I can try to tweak to avoid collapsing large regions like this? 

Best,

Tom

**Comment by chhylp123** [comment]
If one haplotype doesn't have enough coverage, hifiasm cannot separate it. We will expose some options in the new version tomorrow, hopefully it might be helpful. By the way, could you please show the alignment results of corrected HiFi reads? Just curious if two haplotypes have already been collapsed during error correction step.

**Comment by tcb72** [comment]
@chhylp123 

No, the haplotypes do not look to be separated after ec step

![image](https://user-images.githubusercontent.com/29762255/113952066-28e5b080-97d2-11eb-8f9f-f1d690edc345.png)

**Comment by StevenBai97** [comment]
Hi,
I also meet the similar problem. There are several blank crosses in the contigs, and I don't know the reason. The average coverage of genome is about 68X. Should I break the contig and remove the blank ones (like showing in pictures)? If so, will some genes be lost in the genome ?
Look forward to your suggestions. Thank you.
![image](https://user-images.githubusercontent.com/80459240/114024791-0bcee180-98a7-11eb-9655-12ccda4789e9.png)
![image](https://user-images.githubusercontent.com/80459240/114024798-0e313b80-98a7-11eb-996c-9f2d96908fee.png)

**Comment by lh3** [comment]
@tcb72 I agree hifiasm is likely collapsing that region. However, how to resolve that depends not only on the three reads you show in a 790bp region but also on the overlaps they have with reads from the correct haplotype. If the lack of coverage extends to a long region, hifiasm can do little. Also, the coverage on the correct haplotype is very low. It is not 90/10. It is 3 out of ~48 reads, or 6.3%. If hifiasm aggressively resolves this case, it may introduce new problems elsewhere. For example, it is not uncommon to see somatic mutations like this. It is hard. As Haoyu said, he will expose more internal parameters, but it is likely that you have to either increase coverage or manually fix by yourself. You can also try HiCanu to see if it can do a better job.

@StevenBai97 Yours is a distinct problem. The large white band is probably repetitive regions which Hi-C reads can't be mapped to uniquely. It is fine. I am not sure about the smaller one. Please create a new issue. It is confusing to have multiple problems discussed in one github issue.

**Comment by tcb72** [comment]
@StevenBai97 I agree with @lh3... in my experience, those large white bands are large repeat sections. Do not remove them!
